mypy/build.py

```diff
@@ -2128,7 +2133,10 @@ def dispatch(sources: List[BuildSource], manager: BuildManager) -> Graph:
     if manager.options.dump_graph:
         dump_graph(graph)
         return graph
-    process_graph(graph, manager)
+    if manager.options.use_fine_grained_cache:
```

Comment:
Add comment here describing why we have two different ways of processing the cache.